packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -211,135 +209,7 @@ export const useField = <
-        date: TDate | null,
-        boundaries: FieldBoundaries<TDate, TSection>,
-      ) => {
-        const sectionBoundaries = boundaries[activeSection.dateSectionName](date, activeSection);
-
-        // Remove the trailing `0` (`01` => `1`)
-        let newSectionValue = Number(`${activeSection.value}${keyPressed}`).toString();
-
-        while (newSectionValue.length > 0 && Number(newSectionValue) > sectionBoundaries.maximum) {
-          newSectionValue = newSectionValue.slice(1);
-        }
-
-        // In the unlikely scenario where max < 9, we could type a single digit that already exceeds the maximum.
-        if (newSectionValue.length === 0) {
-          newSectionValue = sectionBoundaries.minimum.toString();
-        }
-
-        if (!activeSection.hasTrailingZeroes) {
-          return newSectionValue;
-        }
-
-        return cleanTrailingZeroInNumericSectionValue(newSectionValue, sectionBoundaries.maximum);
-      };
-
-      updateSectionValue({
-        activeSection,
-        setSectionValueOnDate: (activeDate, boundaries) => {
-          // TODO: Support digit editing for months displayed in full letter
-          if (activeSection.contentType === 'letter') {
-            return activeDate;
-          }
-
-          return applySectionValueToDate({
-            utils,
-            dateSectionName: activeSection.dateSectionName,
-            date: activeDate,
-            getNumericSectionValue: (getter) => {
-              const sectionValueStr = getNewSectionValueStr(activeDate, boundaries);
-
-              // We can't parse the day on the current date, otherwise we might try to parse `31` on a 30-days month.
-              // So we take for granted that for days, the digit rendered is always 1-indexed, just like the digit stored in the date.
-              if (activeSection.dateSectionName === 'day') {
-                return Number(sectionValueStr);
-              }
-
-              // The month is stored as 0-indexed in the date (0 = January, 1 = February, ...).
-              // But it is often rendered as 1-indexed in the input (1 = January, 2 = February, ...).
-              // This parsing makes sure that we store the digit according to the date index and not the input index.
-              const sectionDate = utils.parse(sectionValueStr, activeSection.formatValue)!;
-              return getter(sectionDate);
-            },
-            // Meridiem is not compatible with digit editing, this line should never be called.
-            getMeridiemSectionValue: () => '',
-          });
-        },
-        setSectionValueOnSections: (boundaries) => {
-          // TODO: Support digit editing for months displayed in full letter
-          if (activeSection.contentType === 'letter') {
-            return activeSection.value;
-          }
-
-          return getNewSectionValueStr(null, boundaries);
-        },
-      });
-    }
-    // TODO: Improve condition
-    else if (['/', ' ', '-'].includes(keyPressed)) {
```

Comment:
I removed this logic
React Spectrum does not have it either

The only use case where it could be useful is if somebody pressed `1` then `/`.
Before, it would have moved to the next section
Now it stays on the current one

I can't just keep it as it is, otherwise, typing `12/` would move twice to the next section.

I can re-enable it only when the query is on the current section and not empty.
Not sure the complexity is worth it.